api/v1beta1/disruption_cron_types.go

```diff
@@ -83,4 +83,14 @@ type DisruptionCronStatus struct {
 	// Time when the target resource was previously missing.
 	// +nullable
 	TargetResourcePreviouslyMissing *metav1.Time `json:"targetResourcePreviouslyMissing,omitempty"`
+
+	History []DisruptionRun `json:"history,omitempty"`
+}
+
+const MaxHistoryLen = 5
+
+type DisruptionRun struct {
+	Name      string      `json:"name,omitempty"`
```

Comment:
can we change names / kinds between runs?

Reply:
i'm not 100% sure here. Afaiu k8s doesnt have any guarantee that object will not change, or forbid changing. I don't see other logic in our codebase regarding safeguarding these changes. 

is there a quick test i can do with `k edit` to verify this? Do you know how?

Reply:
We identify disruptions per name/namespace so to me it wouldn't make sense we'd be authorized to edit the name field, but i've never really worked with disruption crons before so I don't know if it'd be possible unfortunately...
I asked this since we store the name and kind in the disruption run here ; if they are not editable / they don't change throughout runs, I don't know if it's really useful to keep this info.

Reply:
Also I don't know how 😅 sorry

Reply:
I can say something wrong, but FWIK disruption are immutable _(to reduce the complexity I supposed)_. The only exception is for the `count` field which dans be updated.

Reply:
Disruptions are but I'm unsure about disruption crons 🤔

Reply:
Good question. For cron disruption, I am not aware about how it really works.

Reply:
> can we change names / kinds between runs?

At the very least, we need to support changing names between runs because otherwise, anytime we change the cron --> disruption naming logic, we'd break every existing cron

Reply:
As far as I can understand, this struct is expected to represent a past run of a disruption stored in the status of the disruption cron resource. Changing the name or the spec of the disruption cron resource should not change what is stored in the status since they are past events anyway.

So I don't think there's anything else to be done here? Eventually renaming the struct if not explicit enough.